apps/labs/posts/automated-accessibility-testing-with-bokeh.md

```diff
@@ -0,0 +1,283 @@
+environment to allow axe-core to run. Axe-core required core modules from Node,
+and Bokeh had a stripped away version of node for the most part. I tend to like
+solving problems by myself with minimal hand-holding, but this was one of those
+scenarios that I just couldn't seem to fix without asking for help. The ways
+around the problem looked even more problematic, and I found myself questioning
+my abilities. For this problem, experience did make all the difference and my
+mentor helped me with a part of the codebase we could draw inspiration from. In
+one of our pair programming sessions, he said for us to look more closely at all
+the directories within the Javascript test directory.
+
+We found out that I had been looking at the test setup the wrong way. The bokeh
+test folder has the following [sub
+folders](https://docs.bokeh.org/en/latest/docs/dev_guide/testing.html#select-specific-bokehjs-tests):
+`baselines`, `codebase`, `defaults`, `devtools`, `integration`, and `unit`. The
+`unit`, `defaults`, and `integration` categories of tests were designed to be
+able to both run in the terminal, and on the devtools server. The `unit` and
+`integration` tests looked the most like tests that one would normally see
+and/or write in different projects because they had `describe` blocks and
+`assert` statements. Basing this assumption on just those two facts, I felt the
+location for `playwright` tests would have to model either of those two, to run
+on both terminal & devtools, as well as with assertions. It was on closer
+inspection with my mentor in a pair programming session that we discovered that
+the `codebase` sub-directory used core node modules, and we decided to base
+playwright off that, but imported the necessary parts from `unit` and
+`integration` that would let the assertions and descriptions work as planned.
+This last part had to be done because Bokeh has its own custom-built test
+tooling, similar in syntax to [Jest](https://jestjs.io/), but implemented
+differently. We tested to see if things worked, and they did. We cleaned up the
+code, committed, pushed, and both stood GitHub PR comment section.
+
+On occasion, a question would come up in the PR comment section that would seem
+a bit confusing to me, but then he would clarify what the person was trying to
+ask/communicate, giving me tips on how to implement the suggestions from
+reviewers. It was at that point in time I could really say: "We'd done something
+truly amazing for Bokeh."
+
+## Where we currently are
+
+A testing environment needs tests, and to test what we'd done, we needed a
+variety of valid data visualizations. Luckily, Bokeh very generously had over
+100 example plots in the documentation that we could test against. Testing
+against the documentation came with a few challenges: one of which was the fact
+that the documentation would always need to be built and served locally before
+we could run axe-core tests against it.
```

Comment:
It's frustrating for readers if you introduce a challenge but don't address it

Reply:
That's a very fair point. I can remove that part.